### DISCUSSION TITLE
Backport multi-cargo ships from JGRPP?

### DISCUSSION THREAD

**Discussion by andythenorth** [OpenTTD/OpenTTD#12371] [Features]
JGRPP includes newgrf support for multi-cargo ships. This would be desirable for vanilla.

The implementation is to add 'articulated parts' to the vehicle; these are not articulated with respect to movement or sprites, and function similar to aircraft holds, but with an arbitrary number possible (as I understand it). 

Relevant commits I could find, reverse order (there may be more):

- https://github.com/JGRennison/OpenTTD-patches/commit/7c28ee0213a7e27c33e5e2264bd3b1d79cd02252
- https://github.com/JGRennison/OpenTTD-patches/commit/94cffc0aea6d0744ce9611b7cd9a29442b5d4d04
- https://github.com/JGRennison/OpenTTD-patches/commit/b49a38bf6c5acbf33cb3dc0143e9eea16c2930a7
- https://github.com/JGRennison/OpenTTD-patches/commit/9f5555041782d22a2c20b894c3e53c8fb05473e4
- https://github.com/JGRennison/OpenTTD-patches/commit/c5f026ba663b7d3fb1e098a39a4d593900872cf5
- https://github.com/JGRennison/OpenTTD-patches/commit/211fdd62b295e3c4104f5a1b4146b7031ec7c069

NML patches

- https://github.com/JGRennison/nml/commit/f0c2f84ca6133109a7a623e5b94f805c34ee9769
- https://github.com/JGRennison/nml/commit/229306a185644524b710695bdaf3e7c5c1c5ff0c

JGR's experimental version of multi-SHARK grf 

- https://github.com/JGRennison/Multi-SHARK

Unsinkable Sam test branch for multi-cargo ships 

- https://github.com/andythenorth/unsinkable-sam/tree/articulated-ships

**Comment by JGRennison** [comment]
The GRF documentation for the above is here: [NML](https://jgrennison.github.io/OpenTTD-patches/newgrf-additions-nml.html#ship-callbacks) [NFO](https://jgrennison.github.io/OpenTTD-patches/newgrf-additions.html#callbacks_ships)

The main/initial commit is this one: https://github.com/JGRennison/OpenTTD-patches/commit/211fdd62b295e3c4104f5a1b4146b7031ec7c069.

**Comment by cmcaine** [comment]
A refit order on an articulated ship refits the entire ship as of JGRPP v65.1. It would be good if this was made consistent (so you can refit different parts of the ship in orders, as well as manually when the ship is in a depot).
